lib.js

```diff
@@ -41,36 +45,98 @@ const handleMessages = ( self, ms ) => {
 
 const reduceHtml = m => {
     const mp = m;
-    // <p>
-    //mp.message = m.message.substring( 3, m.message.length-5 );
+
+    const unescapeHtml = unsafe => {
+        return unsafe
+            .replace( /&amp;/g,  `&` )
+            .replace( /&lt;/g,   `<` )
+            .replace( /&gt;/g,   `>` )
+            .replace( /&quot;/g, `"` )
+            .replace( /&#39;/g,  `'` );
+    };
+
+    // <p></p>
+    mp.message = mp.message.replace( /<\/?p[\w =#"':\/\\.\-?]*>/gi, "" );
 
     // <a>gets left</a>
-    mp.message = mp.message.replace( /<\/?a[\w ="':\/\\.]*>/gi, "" );
+    // Custom links are text in <a>, and then the link in <kbd>
+    mp.message = mp.message.replace( /<\/?a[\w -=#"':\/\\.\-?]*>/gi, "" );
+    mp.message = mp.message.replace( "<kbd>", "" );
+    mp.message = mp.message.replace( "</kbd>", "" );
 
     // <img> to :emotes:
-    //mp.message = mp.message.replace( /<img[\w ="':\/\\.\-?]*>/gi, m => {
-    //    return m.search( /https:\/\/[\/\w\.\-?]*/ );
-    //});
+    mp.message = mp.message.replace( /<img[\w -=#"':\/\\.\-?]*>/gi, m => {
+        // (((, ))), :), :( are rendered differently
+        // (They dont even show up in the emote list)
+        //
+        // It wouldn't be so bad if the two echos were 'echol' and 'echor', but
+        //  one echo is flipped with CSS.
+        if( m.includes('alt="echo"') ) {
+            return m.includes('scaleX(-1)') ? "(((" : ")))";
+        }
+        return m.match( /alt="([\w:()]+)"/ )[1];
+    });
+
+    mp.message = turndownService.turndown( mp.message );
```

Comment:
`turndown` doesn't map to [bitwave.tv] chat 1:1, so some pre-filtering is needed